cocotb/triggers.py

```diff
@@ -194,7 +195,21 @@ def __init__(self, time_ps, units=None):
 
         See Also:
             :func:`~cocotb.utils.get_sim_steps`
+
+        Raises:
+            TriggerException: If a negative value is passed for Timer setup.
+
+        .. versionchanged:: 1.5
+            Raise an exception when Timer uses a negative value as it is undefined behavior.
+            Warn for 0 as this will cause erratic behavior in some simulators as well.
         """
+        if time_ps <= 0:
+            if time_ps == 0:
+                warnings.warn("Timer setup with value 0, which might exhibit undefined behavior in some simulators",
+                              category=RuntimeWarning,
+                              stacklevel=2)
+            else:
+                raise TriggerException("Timer value time_ps must not be negative")
         GPITrigger.__init__(self)
```

Comment:
If you don't call `GPITrigger.__init__` first, then you get a weird exception in `Trigger.__del__`
```suggestion
        GPITrigger.__init__(self)
        if time_ps <= 0:
            if time_ps == 0:
                warnings.warn("Timer setup with value 0, which might exhibit undefined behavior in some simulators",
                              category=RuntimeWarning,
                              stacklevel=2)
            else:
                raise TriggerException("Timer value time_ps must not be negative")
```

Reply:
ah, alright. That makes sense. (not easy to spot though for someone non-familiar)